File: lua/src/commons.lua
-- Copyright (C) 2026 Lenik <zephyr@bodz.net>
-- SPDX-License-Identifier: AGPL-3.0-or-later
--
-- Template example: shared helpers live in src/commons.* (not a real module name).
-- In a concrete program, rename to something specific (e.g. stream_copy.lua).

local commons = {}

function commons.copy_stream(src, dst)
    while true do
        local chunk = src:read(8192)
        if not chunk then
            return
        end
        dst:write(chunk)
        dst:flush()
    end
end

function commons.copy_file(path, out)
    local fh, err = io.open(path, "rb")
    if not fh then
        error(err, 0)
    end
    local ok, copy_err = pcall(function()
        commons.copy_stream(fh, out)
    end)
    fh:close()
    if not ok then
        error(copy_err, 0)
    end
end

return commons

File: kotlin/src/commons.kt
/*
 * Copyright (C) 2026 Lenik <zephyr@bodz.net>
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import java.io.IOException
import java.io.InputStream
import java.io.OutputStream
import java.nio.file.Files
import java.nio.file.Path

object commons {
    @Throws(IOException::class)
    fun copyStream(`in`: InputStream, out: OutputStream) {
        val buf = ByteArray(8192)
        while (true) {
            val n = `in`.read(buf)
            if (n == -1) {
                break
            }
            out.write(buf, 0, n)
        }
        out.flush()
    }

    @Throws(IOException::class)
    fun copyFile(file: Path, out: OutputStream) {
        Files.newInputStream(file).use { input ->
            copyStream(input, out)
        }
    }
}
